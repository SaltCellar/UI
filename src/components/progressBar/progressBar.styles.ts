import { StyleSheet } from '@patternfly/react-styles';
import {
  global_spacer_xs,
  global_warning_color_100,
} from '@patternfly/react-tokens';

export const styles = StyleSheet.create({
  progressBar: {
    height: 25,
    padding: global_spacer_xs.value,
    border: '1px solid black',
  },
  bar: {
    height: '100%',
    backgroundColor: global_warning_color_100.value,
  },
});
