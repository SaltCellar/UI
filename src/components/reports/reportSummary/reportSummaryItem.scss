.reportSummaryItem {
  margin-bottom: 10px;
}
