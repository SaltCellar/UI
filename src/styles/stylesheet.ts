import { css, StyleSheet } from 'aphrodite';

const create = StyleSheet.create;
const classNames = (...classes: (string | null | void)[]) => {
  return classes.filter(Boolean).join(' ');
};

export { css, create, classNames };
