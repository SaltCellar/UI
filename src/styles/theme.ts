import {
  global_BackgroundColor_100,
  global_BoxShadow_sm_right,
  global_breakpoint_lg,
  global_breakpoint_md,
  global_breakpoint_sm,
  global_breakpoint_xl,
  global_breakpoint_xs,
  global_Color_dark_100,
  l_page__sidebar_Width,
} from '@patternfly/react-tokens';

const createBreakpoint = (size: string) => `@media (min-width: ${size})`;

const breakpoints = {
  xs: createBreakpoint(global_breakpoint_xs.value),
  sm: createBreakpoint(global_breakpoint_sm.value),
  md: createBreakpoint(global_breakpoint_md.value),
  lg: createBreakpoint(global_breakpoint_lg.value),
  xl: createBreakpoint(global_breakpoint_xl.value),
};

export const theme = {
  breakpoints,
  page_breakpoint: breakpoints.md,
  page_masthead_height: 60,
  page_sidebar_background: global_BackgroundColor_100.value,
  page_sidebar_boxShadow: global_BoxShadow_sm_right.value,
  page_sidebar_width: l_page__sidebar_Width.value,
  verticalNav_color: global_Color_dark_100.value,
  verticalNav_link_color: global_Color_dark_100.value,
};
