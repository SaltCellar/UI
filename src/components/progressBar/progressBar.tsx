import { css } from '@patternfly/react-styles';
import React from 'react';
import { styles } from './progressBar.styles';

interface ProgressBarProps {
  progress: number;
}

const ProgressBar: React.SFC<ProgressBarProps> = ({ progress }) => {
  const width = Math.min(Math.max(0, progress), 100); // force between 0 - 100;
  return (
    <div className={css(styles.progressBar)}>
      <div className={css(styles.bar)} style={{ width: `${width}%` }} />
    </div>
  );
};

export { ProgressBar, ProgressBarProps };
