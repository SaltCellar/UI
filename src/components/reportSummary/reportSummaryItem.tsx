import { css } from '@patternfly/react-styles';
import React from 'react';
import { FormatOptions, ValueFormatter } from 'utils/formatValue';
import { ProgressBar } from '../progressBar';
import { styles } from './reportSummaryItem.styles';

interface ReportSummaryItemProps {
  label: React.ReactText;
  units: string;
  value: number;
  totalValue: number;
  formatValue: ValueFormatter;
  formatOptions?: FormatOptions;
}

const ReportSummaryItem: React.SFC<ReportSummaryItemProps> = ({
  label,
  value,
  totalValue,
  formatValue,
  units,
  formatOptions,
}) => {
  const percent = (value / totalValue) * 100;
  return (
    <li className={css(styles.reportSummaryItem)}>
      <div className={css(styles.info)}>
        <div>{label}</div>
        <div>
          {formatValue(value, units, formatOptions)} ({percent.toFixed(2)}%)
        </div>
      </div>
      <ProgressBar progress={percent} />
    </li>
  );
};

ReportSummaryItem.defaultProps = {
  formatValue: v => v,
};

export { ReportSummaryItem, ReportSummaryItemProps };
