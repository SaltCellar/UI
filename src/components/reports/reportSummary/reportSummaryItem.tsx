import { Progress, ProgressSize } from '@patternfly/react-core';
import React from 'react';
import { injectIntl, WrappedComponentProps } from 'react-intl';
import { FormatOptions, ValueFormatter } from 'utils/formatValue';
import { unitLookupKey } from 'utils/formatValue';
import { reportSummaryItem } from './reportSummaryItem.styles';

interface ReportSummaryItemProps extends WrappedComponentProps {
  formatValue: ValueFormatter;
  formatOptions?: FormatOptions;
  label: string;
  totalValue: number;
  units: string;
  value: number;
}

const ReportSummaryItemBase: React.SFC<ReportSummaryItemProps> = ({
  label,
  formatOptions,
  formatValue,
  intl,
  totalValue,
  units,
  value,
}) => {
  const lookup = unitLookupKey(units);
  const unitsLabel =
    lookup !== 'usd'
      ? intl.formatMessage({ id: `units.${lookup}` })
      : undefined;

  const percent = !totalValue ? 0 : (value / totalValue) * 100;
  const percentVal = Number(percent.toFixed(2));
  const percentLabel = intl.formatMessage(
    { id: 'percent_of_total' },
    {
      percent: percentVal,
      units: unitsLabel,
      value: formatValue(value, units, formatOptions),
    }
  );

  return (
    <li className={reportSummaryItem}>
      <Progress
        label={percentLabel}
        value={percentVal}
        title={label}
        size={ProgressSize.sm}
      />
    </li>
  );
};

ReportSummaryItemBase.defaultProps = {
  formatValue: v => v,
};

const ReportSummaryItem = injectIntl(ReportSummaryItemBase);

export { ReportSummaryItem, ReportSummaryItemProps };
