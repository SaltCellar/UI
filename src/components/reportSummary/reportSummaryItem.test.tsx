import { css } from '@patternfly/react-styles';
import { shallow } from 'enzyme';
import React from 'react';
import { ProgressBar } from '../progressBar';
import { ReportSummaryItem, ReportSummaryItemProps } from './reportSummaryItem';
import { styles } from './reportSummaryItem.styles';

const props: ReportSummaryItemProps = {
  label: 'Label',
  totalValue: 1000,
  value: 100,
  formatValue: jest.fn(),
};

test('fromats value', () => {
  shallow(<ReportSummaryItem {...props} />);
  expect(props.formatValue).toBeCalledWith(props.value);
});

test('gets percentage from value and total value', () => {
  const view = shallow(<ReportSummaryItem {...props} />);
  expect(view.find(ProgressBar).props().progress).toMatchSnapshot();
  expect(view.find(`.${css(styles.info)}`)).toMatchSnapshot();
});
