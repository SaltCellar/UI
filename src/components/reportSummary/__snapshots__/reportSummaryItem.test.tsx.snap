// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets percentage from value and total value 1`] = `10`;

exports[`gets percentage from value and total value 2`] = `
<div
  className="info_sksau8"
>
  <div>
    Label
  </div>
  <div>
     (
    10.00
    %)
  </div>
</div>
`;
